src/RpcServer/RpcServer.Wallet.cs

```diff
@@ -348,20 +347,20 @@ private JObject GetVerificationResult(UInt160 scriptHash, ContractParameter[] ar
             {
                 throw new RpcException(-100, "Unknown contract");
             }
-            var methodName = "verify";
+            var md = contract.Manifest.Abi.GetMethod("verify", 0);
+            if (md is null)
+                throw new RpcException(-101, $"The smart contract {contract.Hash} haven't got verify method without arguments");
+            if (md.ReturnType != ContractParameterType.Boolean)
+                throw new RpcException(-102, "The verify method doesn't return boolean value.");
 
             Transaction tx = signers == null ? null : new Transaction
```

Comment:
Can we also fill the transaction when `signers == null`? We can do it because we know exactly at least one signer (it's the contract to be called). And we also know the witness for this signer: invocation script pushes `args` on the stack; verification script is empty because it's a contract witness.

Reply:
Ok.